template/circle.yml

```diff
@@ -6,7 +6,8 @@ machine:
 
 checkout:
   post:
-    - git fetch origin gh-pages --depth 10
+    - if [ -e .git/shallow ]; then git fetch origin --unshallow; fi
+    - git fetch origin gh-pages
```

Comment:
Oh, and while we could just move the --unshallow fetch there, that's after the cache has snapped a copy of the .git directory, so we'd be unshallowing the repo every run, rather than only once per repo this way.